impl/fs/ext2/cogent/lib/debug.cogent.h

```diff
@@ -0,0 +1,85 @@
+-- Hey, Emacs!  This isn't actually C; it's really -*- cogent -*-.
```

Comment:
I think the long-term plan is to factor out the common infrastructures to `cogent/lib/gum`, and this file looks a good candidate. Do you think it's doable?

Reply:
Yes, I suspect it would. Coming up with a (better?) name for the symbol that this file uses as its guards is a good idea to do at roughly the same time.

(I'm also vaguely interested in seeing `libgum` be a bit more platform-agnostic; currently it relies very heavily on Linux KPI-isms, which are great if Linux is the target, but if, say, we want to target other environments (like, say, a hosted-C environment in a userland, or a fully freestanding C environment, or somewhere in between) this would need a lot of nitpicking.)

Reply:
We used to have a userland target around 2016, mostly for debugging. Later as the file system implementations evolved, no one tried to maintain it and then we left it only in git history.